bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -269,8 +269,8 @@ void ReplicaImp::onMessage<ClientRequestMsg>(ClientRequestMsg *m) {
                                                         << "], senderId=" << senderId);
         if (time_to_collect_batch_ == MinTime) time_to_collect_batch_ = getMonotonicTime();
         requestsQueueOfPrimary.push(m);
-        primary_queue_size_.Get().Set(requestsQueueOfPrimary.size());
         primaryCombinedReqSize += m->size();
+        primary_queue_size_.Get().Set(requestsQueueOfPrimary.size());
```

Comment:
Why did you change the order of operations?

Reply:
I am trying to follow the following conventions:
1) Whenever we push into requestsQueueOfPrimary, we update the (byte)size in  primaryCombinedReqSize.
2) Whenever we pop from the requestsQueueOfPrimary we do the following:
    a) Access the front() of the queue in some variable say 'f' (its a shared ptr to CRM).
    b) Update the (byte)size in  primaryCombinedReqSize.
    c) Pop the queue.
    d) delete the shared ptr.